cloudferrylib/os/actions/check_needed_compute_resources.py

```diff
@@ -61,21 +61,26 @@ def check_in_use_flavor(self, objs, info):
         # is in this flavor, it will keep the old flavor id, can not be matched
         # to existing flavors
         src_compute = self.src_cloud.resources[utl.COMPUTE_RESOURCE]
-        src_flavors = [flavor.id for flavor in src_compute.get_flavor_list()]
+        src_flavor_ids = [flavor.id for flavor in src_compute.get_flavor_list()]
         dst_compute = self.dst_cloud.resources[utl.COMPUTE_RESOURCE]
-        dst_flavors = [flavor.id for flavor in dst_compute.get_flavor_list()]
+        dst_flavors = dst_compute.get_flavor_list()
+        dst_flavor_ids = [flavor.id for flavor in dst_flavors]
 
         for instance in objs.values():
-            if instance['instance']['flavor_id'] not in src_flavors \
-                    and instance['instance']['flavor_id'] not in dst_flavors:
+            inst_flavor_id = instance['instance']['flavor_id']
+            if inst_flavor_id not in src_flavor_ids \
+                    and inst_flavor_id not in dst_flavor_ids:
```

Comment:
I think that:
flavor_ids = src_flavor_ids+dst_flavor_ids
and 
if inst_flavor_id not in flavor_ids: